packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -0,0 +1,221 @@
+import { IResource, Resource, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { AcceptLanguage } from './common';
+import { InputValidator } from './private/validation';
+import { CfnCloudFormationProduct } from './servicecatalog.generated';
+import { Template } from './template';
+
+/**
+ * A Service Catalog product, currently only supports type CloudFormationProduct
+ */
+export interface IProduct extends IResource {
+  /**
+   * The ARN of the product.
+   * @attribute
+   */
+  readonly productArn: string;
+
+  /**
+   * The id of the product
+   * @attribute
+   */
+  readonly productId: string;
+}
+
+/**
+ * A reference to a Service Catalog product.
+ */
+export interface ProductAttributes {
+  /**
+   * The ARN of the product.
+   */
+  readonly productArn: string;
+
+  /**
+   * The name of the product.
+   */
+  readonly productName: string;
+}
+
+abstract class ProductBase extends Resource implements IProduct {
+  public abstract readonly productArn: string;
+  public abstract readonly productId: string;
+}
+
+/**
+ * Properties of provisioning artifacts
+ */
+export interface ProvisioningArtifactProperties {
+  /**
+   * The description of the provisioning artifact
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * Disable template validation.
+   * @default True
+   */
+  readonly disableTemplateValidation?: boolean;
```

Comment:
Can we actually switch this property to have an "active" name, instead of a "disabled" name?

```suggestion
  readonly validateTemplate?: boolean;
```